### PR TITLE
Adding users service

### PR DESCRIPTION
**Pull request by drewctate**

### CODE REVIEW

File: src/app/shared/users/users.service.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@angular/core';
+import { Http, Response } from '@angular/http';
+import { Observable } from 'rxjs/Rx';
+import { User } from './user.model';
+
+@Injectable()
+export class UsersService {
+  private apiUrl = 'http://localhost:3000';
```
Comment (gshigeto) [inline src/app/shared/users/users.service.ts]: Should this be `http://localhost:3000/users` or kept the same?

Comment (mountain01) [inline src/app/shared/users/users.service.ts]: I think that it should be changed.

File: src/app/shared/users/user.model.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export interface User {
+  _id?: string;
+  userId: number;
+  username: string;
+  email: string;
+  dateCreated: Date;
+  profilePhoto: string;
+  refreshToken: string;
```
Comment (jbw91) [inline src/app/shared/users/user.model.ts]: @drewctate This should be optional. It won't always be there.

File: src/app/shared/users/user.model.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export interface User {
+  _id?: string;
+  userId: number;
+  username: string;
+  email: string;
+  dateCreated: Date;
```
Comment (jbw91) [inline src/app/shared/users/user.model.ts]: I'd say make this optional too @drewctate

**Pull request merged**